src/cluster.c

```diff
@@ -98,6 +104,9 @@ static void clusterBuildMessageHdr(clusterMsg *hdr, int type, size_t msglen);
 #define dictEntryPrevInSlot(de) \
     (((clusterDictEntryMetadata *)dictEntryMetadata(de))->prev)
 
+#define isTLSClient(c) (c->conn && (c->conn->type == connectionTypeTls()))
+#define usePort(c) ((!!server.tls_cluster) == isTLSClient(c))
```

Comment:
Maybe `usePrimaryPort` is even more clear? Because both ports are ports.

Just a suggestion: You can use `static inline` function instead of a macro. It is efficient like a macro but gives better type checking.